From 7082a1ec40299025cbbb5b2e82300d4e5a81f6db Mon Sep 17 00:00:00 2001
From: emcmanus <emcmanus@google.com>
Date: Mon, 25 Jun 2018 08:20:19 -0700
Subject: [PATCH 3/4] Adjust AutoValueTest so that it covers setting an
 immutable collection of T from a Collection<? extends T>. The actual bug was
 fixed long ago but never closed and never covered by an explicit test.

RELNOTES=n/a

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=201956399
---
 .../src/test/java/com/google/auto/value/AutoValueTest.java        | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/value/src/it/functional/src/test/java/com/google/auto/value/AutoValueTest.java b/value/src/it/functional/src/test/java/com/google/auto/value/AutoValueTest.java
index fbd378a..b2ed4be 100644
--- a/value/src/it/functional/src/test/java/com/google/auto/value/AutoValueTest.java
+++ b/value/src/it/functional/src/test/java/com/google/auto/value/AutoValueTest.java
@@ -2189,13 +2189,13 @@ public class AutoValueTest {
   public abstract static class BuilderWithCopyingSetters<T extends Number> {
     public abstract ImmutableSet<? extends T> things();
 
-    public abstract ImmutableList<String> strings();
+    public abstract ImmutableList<Number> numbers();
 
     public abstract ImmutableMap<String, T> map();
 
     public static <T extends Number> Builder<T> builder(T value) {
       return new AutoValue_AutoValueTest_BuilderWithCopyingSetters.Builder<T>()
-          .setStrings(ImmutableSet.of("foo", "bar"))
+          .setNumbers(ImmutableSet.of(17, 23.0))
           .setMap(Collections.singletonMap("foo", value));
     }
 
@@ -2207,7 +2207,7 @@ public class AutoValueTest {
 
       Builder<T> setThings(T... things);
 
-      Builder<T> setStrings(Collection<String> strings);
+      Builder<T> setNumbers(Collection<? extends Number> strings);
 
       Builder<T> setMap(Map<String, T> map);
 
@@ -2221,7 +2221,7 @@ public class AutoValueTest {
 
     BuilderWithCopyingSetters<Integer> a = builder.setThings(ImmutableSet.of(1, 2)).build();
     assertEquals(ImmutableSet.of(1, 2), a.things());
-    assertEquals(ImmutableList.of("foo", "bar"), a.strings());
+    assertEquals(ImmutableList.of(17, 23.0), a.numbers());
     assertEquals(ImmutableMap.of("foo", 23), a.map());
 
     BuilderWithCopyingSetters<Integer> b = builder.setThings(Arrays.asList(1, 2)).build();
-- 
2.16.3

